### ISSUE TITLE
Various prebuild issues

### ISSUE THREAD

**Issue by shaal** [gitpod-io/gitpod#6391]
### Bug description

I've been working a lot with prebuilds, and some issues are still not resolved:
 
1. When creating a PR and running a prebuild for the FIRST time, you can go to the Gitpod's URL of the project (ie:  https://gitpod.io/#https://github.com/drud/ddev) and you can see all the commands that the prebuild is running.
But when making new commits after that, and you'll only see "Connecting to workspace logs...".
A prebuild IS RUNNING in the background, but there's no visual representation of what's happening, and when prebuild is done the workspace is reloaded on that browser tab.
Related: https://github.com/gitpod-io/gitpod/issues/6305
![image](https://user-images.githubusercontent.com/22901/138735520-560c903c-9b7c-47ba-8816-f98b861da5ce.png)

1. I set a special webhook to send me an email when a prebuild is running (https://github.com/shaal/DrupalPod/blob/ready-made-envs/.gitpod.yml#L8), every time a prebuild is running, I am getting 2 emails, so for some reason the prebuild is called and running twice, but I don't know why.

1. (Noticeable on projects with `init` start-tasks that take a long time, and perhaps related to the previous issue of prebuild running twice) - 
After manually initiating a prebuild (or pushing a commit), if you go to the Gitpod URL of the project, you will see the Prebuild page (which is expected After Prebuild finish running), and the Gitpod workspace opens, instead the new workspace using the `prebuild` that just finished, it runs all the `init` tasks of the prebuild all over again.
If you close this workspace, and open a new workspace, now prebuild is trully reloaded, and only `command` tasks are running in the workspace.

### Steps to reproduce

1. Fork repo with long `init` tasks (ie. https://github.com/shaal/DrupalPod)
1. Create a commit and push it.
1. Watch the prebuild process (issues 2 and 3) + workspace that opens right after.
1. Create another commit and push it.
1. Watch the prebuild process (issues 1 and 2 and 3) + workspace that opens right after.

### Expected behavior

1. Prebuild should only run once per commit (or manual trigger).
1. Workspace should never run prebuild tasks in the workspace.
1. Prebuild should always display what commands are running while prebuild is running.

**Comment by shaal** [comment]
I noticed that you can see prebuilds are running twice in the UI (on same commit, 2 prebuilds running):
I pushed a commit to a branch, which triggered the 2 simultaneous prebuilds -
![image](https://user-images.githubusercontent.com/22901/141925208-ea62ae90-0018-474a-a75a-56baf8cd631e.png)

**Comment by JanKoehnlein** [comment]
Sorry, as we're a bit swamped ATM, I have to unfortunately move this to our backlog. 

Chances to get this fixed are much better if you file separate GH issues for the individual problems.

**Comment by jankeromnes** [comment]
Many thanks for the in-depth debugging of how prebuilds work! 🙏

Based on https://github.com/gitpod-io/gitpod/issues/6391#issuecomment-977762407, removing temporarily from Groundwork, although I agree we should fix these problems if they still occur.

Additionally, I agree that it would be better to have separate issues for separate problems, in order to know which problems are already solved and which ones still need work.

**Comment by AlexTugarev** [comment]
Hey @shaal!
I retested most, hopefully all of the issues reported here. 

The log related issues were already tackled in separate PRs, and while the situation is still not ideal, the reported issue 1.) was resolved. 

I retested push events on a fork of the repo posted, and added a similar notification to the `init` task like so:

<img width="616" alt="Screen Shot 2022-06-10 at 09 52 48" src="https://user-images.githubusercontent.com/914497/173018248-6f350359-1ea4-4108-878c-c763af45b048.png">

Which turns out to work as expected, the prebuild ran and the endpoint got called a single time:

<img width="980" alt="Screen Shot 2022-06-10 at 09 52 24" src="https://user-images.githubusercontent.com/914497/173019286-b1c326a8-571f-4658-a8d7-cee0caf43af9.png">

With that, I'm going to close this issue here. Let's create smaller ones and solve separately if something else pops up.
